src/utils/archive.ts

```diff
@@ -199,7 +200,59 @@ export async function createPublicDrive(
     txs: [driveTx, rootFolderTx]
   };
 }
+export async function sendArDriveFee(
+  keyfile: JWKInterface,
+  arPrice: number,
+  arweave: Arweave
+) {
+  try {
+    // Get the latest ArDrive Community Fee from the Community Smart Contract
+    let fee = arPrice * ((await getArDriveFee()) / 100);
+
+    // If the fee is too small, we assign a minimum
+    if (fee < 0.00001) {
+      fee = 0.00001;
+    }
+
+    // Probabilistically select the PST token holder
+    const holder = await selectTokenHolder();
+
+    // send a fee. You should inform the user about this fee and amount.
+    const transaction = await arweave.createTransaction(
+      {
+        target: holder,
+        quantity: arweave.ar.arToWinston(fee.toString())
+      },
+      keyfile
+    );
 
+    // Tag file with data upload Tipping metadata
+    transaction.addTag("App-Name", ardriveClient);
+    transaction.addTag("App-Version", ardriveVersion);
+    transaction.addTag("Type", "fee");
+    transaction.addTag("Tip-Type", "data upload");
+
+    // Sign file
+    await arweave.transactions.sign(transaction, keyfile);
+
+    // Submit the transaction
+    const response = await arweave.transactions.post(transaction);
+    if (response.status === 200 || response.status === 202) {
+      console.log(
+        "SUCCESS ArDrive fee of %s was submitted with TX %s to %s",
+        fee.toFixed(9),
```

Comment:
Why round down to 9 decimals? AR numbers are valid up to 12 decimal places. With 12 decimal places, I think you're good up to about 900AR, then number precision get sketchy.

Reply:
[Resolved](https://github.com/th8ta/ArConnect/pull/33/commits/66b11719315a0ac4ea310ba6d165889efd900a59)